### DISCUSSION TITLE
Using Piper voices in Firefox offline

### DISCUSSION THREAD

**Discussion by atkirtland** [ken107/read-aloud#411] [Q&A] [answered]
Hi, it seems like the Piper voices require internet access to connect with https://piper.ttstool.com/ (there's a "TypeError: NetworkError when attempting to fetch resource." on this site if using offline), but the speech is generated locally. Why is internet access required? It seems like Piper could be used offline using a setup as described in [this](https://github.com/ken107/read-aloud/issues/375) issue, but is there a simple or more convenient way?

**Comment by ken107** [comment, marked as answer]
We still have to implement this to have fully offline functionality
https://github.com/ken107/piper-browser-extension/issues/2
